runbooks/how-to-load-cloudwatch-historical-data.md

```diff
@@ -0,0 +1,200 @@
+Follow this runbook to load historical production access log data into BFD insights.
+
+There are three Glue tables that are referenced in this runbook:
+- `export` table - a table that is built on top of the raw export data from Cloudwatch
+- `staging` table - a non-partitioned table that has the desired column structure and can be loaded from the `export` table efficiently
+- `target` table - the partitioned table that is the final destination for the data (should already exist via terraform)
+
+1. Export the data from Cloudwatch.
+   1. Review the exports that are available already in the export location: s3://bfd-insights-bfd-app-logs/export/prod/
+      to see if the needed data has been exported previously. New exports should be contiguous and non-overlapping with
+      existing exports. Start and end timestamps should be selected to run from the first day of a month to the last day
+      of a month when possible. Start and end times should be set to 00:00:00 UTC. Removing an existing export in favor
+      of exporting that data again with a later end date is an acceptable way to keep the number of exports manageable. 
+   3. Navigate to Cloudwatch in the AWS console.
+   4. Select `Log Groups`
+   5. Select `/bfd/prod/bfd-server/access.json`
+   6. Select `Actions -> Export Data to Amazon S3`
+   7. Choose the time period 
+   8. Select Account: *This Account*
+   9. S3 Bucket Name: `bfd-insights-bfd-app-logs`
+   10. S3 Bucket Prefix: `export/prod/YYYY_(MM-MM)`
+   11. Run the job. Production exports can take up to an hour per month depending on the activity level that month.
+   12. When the job completes, remove the `aws-logs-write-test` subfolder that AWS creates for internal testing.
+
+2. Create/Update the Glue `export` table.
+   1. From the AWS Glue console, select Crawlers
+   2. Select the `bfd_cw_export` crawler (this crawler is in terraform and should already exist)
+   3. Select `Run`
+   4. Verify that the `bfd_cw_export.prod` table has the expected data by inspecting the results of this query that
+      retrieves the number of records per month and ensuring that the newly exported data is represented:
+   
+   ```sql
+   select date_format(from_iso8601_timestamp("timestamp"), '%Y-%m'), count(*)
+   from bfd_cw_export.prod
+   group by 1
+   order by 1
+   ```
+
+3. Determine the column list for `staging` and `target` tables.
+   1. This step may be skipped if the `target` table already exists. In that case the column list for the `staging` table
+      should be identical in names, data types, and ordering to the `target` table after removing the `year` and `month`
+      partition columns.
+   3. This Athena query produces the canonical ordered list of JSON MDC keys from a Cloudwatch export.
```

Comment:
I think the bullet numbers are maybe messed up for this and the below entry (we go 3.1, 3.3, 3.3)